geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala

```diff
@@ -97,18 +102,116 @@ class TransformSimpleFeature(transformSchema: SimpleFeatureType,
 object TransformSimpleFeature {
 
   import scala.collection.JavaConversions._
+  import scala.collection.JavaConverters._
 
   def apply(sft: SimpleFeatureType, transformSchema: SimpleFeatureType, transforms: String): TransformSimpleFeature = {
     val a = attributes(sft, transformSchema, transforms)
     new TransformSimpleFeature(transformSchema, a)
   }
 
+  def apply(sft: SimpleFeatureType, properties: Seq[String]): TransformSimpleFeature = {
+
+    val (transformSft, transforms) = queryToTransformSFT(sft, properties)
+
+    TransformSimpleFeature(sft, transformSft, transforms)
+  }
+
   def attributes(sft: SimpleFeatureType,
                  transformSchema: SimpleFeatureType,
                  transforms: String): Array[(SimpleFeature) => AnyRef] = {
     TransformProcess.toDefinition(transforms).map(attribute(sft, _)).toArray
   }
 
+  def queryToTransformSFT(sft: SimpleFeatureType, properties: Seq[String]): (SimpleFeatureType, String) = {
```

Comment:
i don't think we should copy this code in here... it's sufficiently complex that we don't want to maintain two copies. It seems simple enough to call the QueryPlanner method if you want to construct a TransformSimpleFeature